src/backend/distributed/deparser/deparse_table_stmts.c

```diff
@@ -273,6 +302,85 @@ AppendAlterTableCmdAddConstraint(StringInfo buf, Constraint *constraint,
 			appendStringInfo(buf, " NO INHERIT");
 		}
 	}
+	else if (constraint->contype == CONSTR_FOREIGN)
+	{
+		appendStringInfoString(buf, " FOREIGN KEY");
+
+		AppendColumnNameList(buf, constraint->fk_attrs);
+
+		appendStringInfoString(buf, " REFERENCES");
+
+		appendStringInfo(buf, " %s", quote_identifier(constraint->pktable->relname));
+
+		if (list_length(constraint->pk_attrs) > 0)
+		{
+			AppendColumnNameList(buf, constraint->pk_attrs);
+		}
+
+		/*
+		 * Append supported options if provided.
+		 * Note: When a constraint name is provided by the client therefore we do not deparse the statement in preprocessing,
+		 * the unsupported statements are caught in postprocessing in ErrorIfUnsupportedForeignConstraintExists function.
+		 * Hence the error messages for the unsupported alter table statements are unexpectedly different for the
+		 * ALTER TABLE ... ADD FOREIGN KEY and ALTER TABLE ...  ADD CONSTRAINT <conname> FOREIGN KEY
+		 */
+
+		/* FKCONSTR_MATCH_SIMPLE is default. Append matchtype if not default */
+		if (constraint->fk_matchtype == FKCONSTR_MATCH_FULL)
+		{
+			appendStringInfoString(buf, " MATCH FULL");
+		}
+
+		switch (constraint->fk_del_action)
+		{
+			case FKCONSTR_ACTION_NOACTION:
+			{
+				appendStringInfoString(buf, " ON DELETE NO ACTION");
+				break;
+			}
+
+			case FKCONSTR_ACTION_RESTRICT:
+			{
+				appendStringInfoString(buf, " ON DELETE RESTRICT");
+				break;
+			}
+
+			case FKCONSTR_ACTION_CASCADE:
+			{
+				appendStringInfoString(buf, " ON DELETE CASCADE");
+				break;
+			}
+
+			default:
+			{
+				elog(ERROR, "unsupported FK delete action type: %d",
+					 (int) constraint->fk_del_action);
+				break;
+			}
+		}
+
+		switch (constraint->fk_upd_action)
+		{
+			case FKCONSTR_ACTION_NOACTION:
+			{
+				appendStringInfoString(buf, " ON UPDATE NO ACTION");
+				break;
+			}
+
+			case FKCONSTR_ACTION_RESTRICT:
+			{
+				appendStringInfoString(buf, " ON UPDATE RESTRICT");
+				break;
+			}
+
+			default:
```

Comment:
Why dont we have FKCONSTR_ACTION_SETDEFAULT, FKCONSTR_ACTION_SETNULL, FKCONSTR_ACTION_CASCADE here?

Reply:
Those are not supported based on multi_foreign_key.sql.  Deparser errors out for those unsupported options.

https://github.com/citusdata/citus/blob/28ed013a91a7a3b3048c716153856580294571f4/src/test/regress/sql/multi_foreign_key.sql#:~:text=foreign%20constraint%20creation-,%2D%2D%20test%20foreign%20constraint%20creation%20with%20not%20supported%20parameters,test_constraint%20FOREIGN%20KEY(ref_id)%20REFERENCES%20referenced_table(id)%20ON%20UPDATE%20CASCADE%3B,-%2D%2D%20test%20foreign

Reply:
That error is happening when referencing columns include DistKey. Can we have other types of foreign keys?

Reply:
Check this out:
``` SQL
-- add coordinator
SELECT 1 FROM master_add_node('localhost', 9700, groupId => 0);

CREATE TABLE citus_local_table(l1 int);
SELECT citus_add_local_table_to_metadata('citus_local_table');
CREATE TABLE reference_table(r1 int primary key);
SELECT create_reference_table('reference_table');

ALTER TABLE citus_local_table ADD FOREIGN KEY(l1) REFERENCES reference_table(r1) ON UPDATE CASCADE;
ERROR:  unsupported FK update action type: 99

-- however the following works
ALTER TABLE citus_local_table ADD CONSTRAINT fkey_local_to_ref FOREIGN KEY(l1) REFERENCES reference_table(r1) ON UPDATE CASCADE;
```

Reply:
Similar to this example, you can do `ON UPDATE SET NULL`, `ON UPDATE SET DEFAULT`
So you should include those in the deparser and add some tests encapsulating these